create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusHandler.java

```diff
@@ -86,6 +95,45 @@ private Reference generateReference(DocTp docTp) {
         return reference;
     }
 
+    private List<TitletextTp> getTitleText(DocTp docTp) {
+        return docTp.getItem().getItem().getBibrecord().getHead().getCitationTitle().getTitletext();
+    }
+
+    private String extractMainTitle(DocTp docTp) {
+        StringBuilder mainTitleBuilder = new StringBuilder();
+        getTitleText(docTp)
+            .stream()
+            .filter(titleTextTp -> titleTextTp.getOriginal().equals(YesnoAtt.Y))
+            .findFirst()
+            .ifPresent(part -> mainTitleBuilder.append(getContentWithXmlTags(part.getContent())));
+        return mainTitleBuilder.toString();
+    }
+
+    private String getContentWithXmlTags(List<Object> contents) {
+        StringBuilder contentStringBuilder = new StringBuilder();
+        contents.forEach(content -> {
+            if (content instanceof String) {
+                contentStringBuilder.append(content);
+            } else if (content instanceof JAXBElement<?>) {
+                var element = (JAXBElement<?>) content;
+                String contentText = getContent(element).stream().map(Object::toString).collect(Collectors.joining());
+                contentStringBuilder.append(String.format(XML_TAG_BUILDER, element.getName(),
+                                                          contentText, element.getName()));
+            }
+        });
+        return contentStringBuilder.toString();
+    }
```

Comment:
We should have a short conversation about this:
 - Not sure I have understood what is happening here
 - Avoid lambdas with method bodies (for readability)
 - Where we have if - else if, in this and the called method, are we ignoring certain cases?